src/lib/entity-state.ts

```diff
@@ -388,6 +396,39 @@ export abstract class EntityState<T> {
     patchState({ error: payload });
   }
 
+  goToPage(
```

Comment:
This method could be simplified after changing the typings.
We'll wait for mark reply to the comment above.
In the meantime, you could do the other requested changes.